app/main/views/suppliers.py

```diff
@@ -86,6 +88,119 @@ def download_agreement_file(supplier_id, framework_slug, document_name):
     return redirect(url)
 
 
+@main.route('/suppliers/<supplier_id>/countersigned-agreements/<framework_slug>/'
+            '<document_name>', methods=['GET'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def list_countersigned_agreement_file(supplier_id, framework_slug, document_name):
+    supplier = data_api_client.get_supplier(supplier_id)['suppliers']
+    framework = data_api_client.get_framework(framework_slug)['frameworks']
+    agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
+    prefix = '{}/agreements/{}/{}-{}'.format(framework['slug'], supplier_id, supplier_id, document_name)
+    countersigned_agreement_document = agreements_bucket.list(prefix=prefix)
+    if countersigned_agreement_document:
+        countersigned_agreement = countersigned_agreement_document[-1]
+        countersigned_agreement['last_modified'] = datetimeformat(parse_date(
+            countersigned_agreement['last_modified']))
+        countersigned_agreement = [countersigned_agreement]
+    else:
+        countersigned_agreement = []
+
+    return render_template(
+        "suppliers/upload_countersigned_agreement.html",
+        supplier=supplier,
+        framework=framework,
+        countersigned_agreement=countersigned_agreement,
+        **get_template_data())
+
+
+@main.route('/suppliers/<supplier_id>/countersigned-agreements/<framework_slug>/'
+            '<document_name>', methods=['POST'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def upload_countersigned_agreement_file(supplier_id, framework_slug, document_name):
+    agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
+    errors = {}
+
+    if request.files.get('countersigned_agreement'):
+        the_file = request.files['countersigned_agreement']
+        if not file_is_pdf(the_file):
+            errors['countersigned_agreement'] = 'not_pdf'
+
+        if 'countersigned_agreement' not in errors.keys():
+            filename = '{}/agreements/{}/{}-countersigned-framework-agreement.pdf'.format(
```

Comment:
This could use the [`get_countersigned_agreement_document_path`](https://github.com/alphagov/digitalmarketplace-utils/blob/master/dmutils/documents.py#L207) function.